src/terrama2/impl/DataAccessorGrADS.cpp

```diff
@@ -175,6 +177,37 @@ void terrama2::core::DataAccessorGrADS::addToCompleteDataSet(std::shared_ptr<te:
   }
 }
 
+std::unique_ptr<te::rst::Raster> terrama2::core::DataAccessorGrADS::adaptRaster(const std::unique_ptr<te::rst::Raster>& raster) const
+{
+  auto oldGrid = raster->getGrid();
+
+  std::vector<te::rst::BandProperty*> bands;
+  for(size_t i = 0; i < raster->getNumberOfBands(); ++i)
+  {
+    bands.push_back(new te::rst::BandProperty(*raster->getBand(i)->getProperty()));
+  }
+  auto grid = new te::rst::Grid(raster->getNumberOfColumns(), raster->getNumberOfRows(), new te::gm::Envelope(*raster->getExtent()), raster->getSRID());
+  std::unique_ptr<te::rst::Raster> expansible(te::rst::RasterFactory::make("EXPANSIBLE", grid, bands, {}));
+
+  for(size_t row = 0; row < grid->getNumberOfRows(); ++row)
+  {
+    for(size_t col = 0; col < grid->getNumberOfColumns(); ++col)
+    {
+      auto coord = grid->gridToGeo(col, row);
```

Comment:
Doesn't need to calculate the coord, just use the inverted row ( row - numberOfRows)